### PR TITLE
feature/CP-5603-app-banners-method-get-app-banners-category-id-ios

### PR DESCRIPTION
**Pull request by cpandya25**
Added support of get app banners from category id.

### CODE REVIEW

File: CleverPush/Source/CPAppBannerModuleInstance.m [hunk 1]
```diff
@@ -196,7 +196,17 @@ - (void)getBanners:(NSString*)channelId bannerId:(NSString*)bannerId notificatio
 
     NSMutableURLRequest* request = [[CleverPushHTTPClient sharedClient] requestWithMethod:HTTP_GET path:bannersPath];
     [CleverPush enqueueRequest:request onSuccess:^(NSDictionary* result) {
-        NSArray *jsonBanners = [result objectForKey:@"banners"];
+        NSMutableArray *jsonBanners = [[NSMutableArray alloc]init];
+        if(categoryId != nil && ![categoryId isEqualToString:@""]) {
+            for(NSDictionary *json in [result objectForKey:@"banners"]) {
```
Comment (gbhrdt) [inline CleverPush/Source/CPAppBannerModuleInstance.m:201]: space after for

File: CleverPush/Source/CPAppBannerModuleInstance.m [hunk 2]
```diff
@@ -196,7 +196,17 @@ - (void)getBanners:(NSString*)channelId bannerId:(NSString*)bannerId notificatio
 
     NSMutableURLRequest* request = [[CleverPushHTTPClient sharedClient] requestWithMethod:HTTP_GET path:bannersPath];
     [CleverPush enqueueRequest:request onSuccess:^(NSDictionary* result) {
-        NSArray *jsonBanners = [result objectForKey:@"banners"];
+        NSMutableArray *jsonBanners = [[NSMutableArray alloc]init];
+        if(categoryId != nil && ![categoryId isEqualToString:@""]) {
```
Comment (gbhrdt) [inline CleverPush/Source/CPAppBannerModuleInstance.m:200]: space after if

File: CleverPush/Source/CPAppBannerModuleInstance.m [hunk 3]
```diff
@@ -196,7 +196,17 @@ - (void)getBanners:(NSString*)channelId bannerId:(NSString*)bannerId notificatio
...
+            for(NSDictionary *json in [result objectForKey:@"banners"]) {
+                if([json objectForKey:@"categoryID"] != nil && [[json objectForKey:@"categoryID"] isEqualToString:categoryId]) {
```
Comment (gbhrdt) [inline CleverPush/Source/CPAppBannerModuleInstance.m:202]: categoryID -> category

File: CleverPush/Source/CPAppBannerModuleInstance.m (hunk 2 above)
Comment (gbhrdt) [reply CleverPush/Source/CPAppBannerModuleInstance.m:200]: Also, please do the whole filtering in an extra function to keep things clean here.

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: CleverPush/Source/CPAppBannerModuleInstance.m (hunk 3 above)
Comment (cpandya25) [reply CleverPush/Source/CPAppBannerModuleInstance.m:202]: @gbhrdt , "**categoryID**" is the key, which will come from the backend (API's response), but okay, let me change it to **category** from **categoryID**.

File: CleverPush/Source/CPAppBannerModuleInstance.m [hunk 4]
```diff
@@ -196,7 +196,15 @@ - (void)getBanners:(NSString*)channelId bannerId:(NSString*)bannerId notificatio
 
     NSMutableURLRequest* request = [[CleverPushHTTPClient sharedClient] requestWithMethod:HTTP_GET path:bannersPath];
     [CleverPush enqueueRequest:request onSuccess:^(NSDictionary* result) {
-        NSArray *jsonBanners = [result objectForKey:@"banners"];
+        NSMutableArray *jsonBanners = [[NSMutableArray alloc] init];
+        if (groupId != nil && ![groupId isEqualToString:@""]) {
+            NSString *expression=[NSString stringWithFormat:@"SELF contains '%@'",groupId];
```
Comment (gbhrdt) [inline CleverPush/Source/CPAppBannerModuleInstance.m:201]: Code style (spaces)

Also why do we need this expression? Can't we directly check for banners which have the field `group`= groupId?

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: CleverPush/Source/CPAppBannerModuleInstance.m (hunk 4 above)
Comment (cpandya25) [reply CleverPush/Source/CPAppBannerModuleInstance.m:201]: Hello @gbhrdt  We are getting the array of **banners**, and from that, we need to filter which array element contains the groupId. That's why we need to do the filter.
Before using filter, I have used a loop for adding the objects with the same **groupId**, but due to less code, I have used an expression for getting the filtered data.

Comment (cpandya25) [reply CleverPush/Source/CPAppBannerModuleInstance.m]: Hello @gbhrdt  We are getting the array of **banners**, and from that, we need to filter which array element contains the groupId. That's why we need to do the filter.
Before using filter, I have used a loop for adding the objects with the same **groupId**, but due to less code, I have used an expression for getting the filtered data.

**Pull request merged**